cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -347,3 +356,25 @@ func clientOptionsBuilder(httpClient httputil.HttpClient, userAgent string) *azs
 		WithTransport(httpClient).
 		WithPerCallPolicy(azsdk.NewUserAgentPolicy(userAgent))
 }
+
+type PurgeImplementation func(ctx context.Context, subscriptionId string, name string, location string) error
+
+func (cli *azCli) getPurgeImplementation(resourceType PurgeResourceType) PurgeImplementation {
```

Comment:
Does this make it harder for us to refactor away from the uber `azCli` struct in favor of individual clients?

Reply:
I'm also not sold on this part of the refactoring - Looking at the callers of this method in `bicep_provider.go` we already have context when we call `PurgeResource` as to which type of resource we want to purge (and it's further enforced by the new `PurgeResource` API you added, since I have to pass in a type) and so it feels to me like we should continue to just have the `AzCli` type expose the resource specific primitives and have the code that decides which of these to call live in the bicep layer, which already has a bunch of other policy decisions.  It also feels like long term the set of information we need to purge a resource will be specific to that type, and so trying to group everything under a single PurgeResource operation will be confusing.

I'm sure there are a bunch of pros off this design, but I'm personally struggling to see them, could you help me understand what pushed you towards this refactor?

Reply:
> I'm sure there are a bunch of pros off this design

Not really!! hahaha... I was just hoping to reduce the `purge` api from the `az` interface to just one. And use that one to call the specific implementation.  This is because I know we will be adding purge support for some more stuff in the future and noticed they all use the same input, just for different clients.

I will revert this, I don't have a strong need to keep it